Enable custom log format

eradman opened the pull request:
Color output is default if RSET_HOST_CONNECT is not defined.

Escape codes are not interpreted, but ANSI escape code may be embedded in the environment variables:

    RSET_HOST_CONNECT=$(printf "\x1b[7m=== %%h ===\x1b[0m")

eradman commented:
@vorakl this is a simple approach to customizing status messages using environment variables, which I feel is more usable than command line arguments, and much simpler than a configuration file.

Let me know if the naming convention makes sense.

vorakl commented:
@eradman, It looks fantastic!

> this is a simple approach to customizing status messages using environment variables

I love this approach. Totally agree, it fits best this scenario. 

> Color output is default if RSET_HOST_CONNECT is not defined.

I think, this needs some clarification in the man page. People might try to rewrite only a label's message and get nothing working which would mistakenly guessed as a bug. In addition, if _RSET_HOST_CONNECT_ is set to an empty line, and consequently,  all other envs get reset, the output would also be a bit confusing. This behavior is clear after reading the code, but it'd be great to also have some explanation in the man. 
```
$ env RSET_HOST_CONNECT="" rset -x env garden

Hi there
Bye everyone
```

In general, I think the logic is good. If one is going to rewrite logging messages, then the connect env would most probably be changed.

I have a suggestion for the _RSET_LABEL_EXEC_ 

An exit status is available only for _RSET_LABEL_EXEC_ERROR_ and  is printed when a label has already failed. Although, _RSET_LABEL_EXEC_ is printed before label execution and thus doesn't have an exit status. But, as long as all these messages are printed by the same function and in the same stdout, there is no conceptual difference between _RSET_LABEL_EXEC_ and _RSET_LABEL_EXEC_ERROR_.

Here is an example:

```
$ env RSET_HOST_CONNECT="%T rset: Connecting host=%h" RSET_LABEL_EXEC="%T rset: Running label=%l host=%h" RSET_LABEL_EXEC_ERROR="%T rset: Failed: laber=%l host=%h exit_code=%e" RSET_HOST_DISCONNECT="%T rset: Disconnected host=%h exit_code=%e" rset -x env garden
2023-11-17 19:08:26-0800 rset: Connecting host=garden
2023-11-17 19:08:29-0800 rset: Running label=env_1st host=garden
Hi there
2023-11-17 19:08:31-0800 rset: Running label=env_2nd host=garden
Bye everyone
2023-11-17 19:08:31-0800 rset: Disconnected host=garden exit_code=2
```

Although **env_1st** returned with exitcode=1 and **env_2nd** with exitcode=2, I don't see that.  But it would be nice to

When I add _-e_:

```
$ env RSET_HOST_CONNECT="%T rset: Connecting host=%h" RSET_LABEL_EXEC="%T rset: Running label=%l host=%h" RSET_LABEL_EXEC_ERROR="%T rset: Failed: laber=%l host=%h exit_code=%e" RSET_HOST_DISCONNECT="%T rset: Disconnected host=%h exit_code=%e" rset -e -x env garden
2023-11-17 19:08:55-0800 rset: Connecting host=garden
2023-11-17 19:08:58-0800 rset: Running label=env_1st host=garden
Hi there
2023-11-17 19:08:59-0800 rset: Failed: laber=env_1st host=garden exit_code=1
2023-11-17 19:08:59-0800 rset: Disconnected host=garden exit_code=1
```

I see exitcode=1 because it it available only in _RSET_LABEL_EXEC_ERROR_. 

I'm thinking about renaming RSET_LABEL_EXEC to RSET_LABEL_EXEC_BEGIN, adding RSET_LABEL_EXEC_END, and the last one will be printed when label is finished and exit code is available. It might print exit code other than 0 and keep running, which is totally legit as there will be no _-e_ option. Although, for a debugging purposes, seeing an exit code is valuable.

_RSET_LABEL_EXEC_ERROR_ is still nice to have as it is used in _-e_ mode and non zero exit status really means a failed state which allows a user to define a specific error message.

eradman commented:
Exelent suggestions!

1. Renamed `RSET_LABEL_EXEC` to `RSET_LABEL_EXEC_BEGIN`
2. Added `RSET_LABEL_EXEC_END`
3. Updated the man page to indicate that `RSET_HOST_CONNECT` disables the default messages and is required for custom status messages

vorakl commented:
Very nice! I like it :)

I've just noticed, when looked at the final  output, there is one confusing situation. While it is totally correct and logical in the _-e_ mode, when rset fails immediately on any error and disconnects with the last non-zero exit code, for the default workflow, it's a bit different. rset ignores exit codes and keeps running. It correctly prints out all the exit codes after each label but, a disconnect message returns the last exit code which might be non zero. Although, it exits with 0. 
That's totally correct, but doesn't make sense.

I think, it needs one one if-statement to check if it's not the _-e_ mode and exit code is not zero, then print exit code 0 in the disconnect message  as it is the rset's exit code  anyway.

Something like:

```
if (socket_path)
{
     log_msg(host_disconnect_msg, hostname, "", stop_on_err_opt ? exit_code : 0);
....
```

Here is an example:

```
$ env RSET_HOST_CONNECT="%T rset: Connecting to host=%h" RSET_LABEL_EXEC_BEGIN="%T rset: Running label=%l host=%h" RSET_LABEL_EXEC_END="%T rset: Finished label=%l host=%h exit_code=%e" RSET_LABEL_EXEC_ERROR="%T rset: ERROR: laber=%l host=%h exit_code=%e" RSET_HOST_DISCONNECT="%T rset: Disconnected host=%h exit_code=%e" rset -x env garden; echo $?
2023-11-18 11:24:16-0800 rset: Connecting to host=garden
2023-11-18 11:24:19-0800 rset: Running label=env_1st host=garden
Hi there
2023-11-18 11:24:20-0800 rset: Finished label=env_1st host=garden exit_code=1
2023-11-18 11:24:20-0800 rset: Running label=env_2nd host=garden
Bye everyone
2023-11-18 11:24:20-0800 rset: Finished label=env_2nd host=garden exit_code=2
2023-11-18 11:24:20-0800 rset: Disconnected host=garden exit_code=2
0
```

```
$ env RSET_HOST_CONNECT="%T rset: Connecting to host=%h" RSET_LABEL_EXEC_BEGIN="%T rset: Running label=%l host=%h" RSET_LABEL_EXEC_END="%T rset: Finished label=%l host=%h exit_code=%e" RSET_LABEL_EXEC_ERROR="%T rset: ERROR: laber=%l host=%h exit_code=%e" RSET_HOST_DISCONNECT="%T rset: Disconnected host=%h exit_code=%e" rset -e -x env garden; echo $?
2023-11-18 11:24:36-0800 rset: Connecting to host=garden
2023-11-18 11:24:39-0800 rset: Running label=env_1st host=garden
Hi there
2023-11-18 11:24:40-0800 rset: Finished label=env_1st host=garden exit_code=1
2023-11-18 11:24:40-0800 rset: ERROR: laber=env_1st host=garden exit_code=1
2023-11-18 11:24:40-0800 rset: Disconnected host=garden exit_code=1
1
```

vorakl commented:
By the way, escape sequences can also be integrated using that print-technique ;)

```
$ env RSET_HOST_CONNECT="$(printf "\n---> Host: %%h")" RSET_LABEL_EXEC_BEGIN="$(printf "\n\t---> Label: %%l\n")" RSET_LABEL_EXEC_END="$(printf "\n\t===> Label: %%l\t[%%e]\n")" RSET_HOST_DISCONNECT="$(printf "\n===> Host: %%h\t[%%e]")" rset -x env garden

---> Host: garden

	---> Label: env_1st
Hi there

	===> Label: env_1st	[1]

	---> Label: env_2nd
Bye everyone

	===> Label: env_2nd	[2]

===> Host: garden	[2]
```

eradman commented:
I apprciate the detailed review! Your suggested change to the disconnect `exit_code` makes sense to me. Change incorporated.
